### ISSUE TITLE
[cosmetic] "may come from an old version of the … crate" suggestion may use "_" instead of "-"

### ISSUE THREAD

**Issue by grinapo** [rust-lang/rust#139545]
### Code

```Rust
not relevant.
```

### Current output

```Shell
warning: unexpected `cfg` condition name: `ruma_unstable_exhaustive_types`
  --> src/pub_banlist.rs:18:48
   |
18 | #[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
   |                                                ^^^^^^^^^^^^
   |
   = help: expected names are: `docsrs`, `feature`, and `test` and 31 more
   = note: using a cfg inside a derive macro will use the cfgs from the destination crate and not the ones from the defining crate
   = help: try referring to `EventContent` crate for guidance on how handle this unexpected cfg
   = help: the derive macro `EventContent` may come from an old version of the **`ruma_macros`** crate, try updating your dependency with `cargo update -p ruma_macros`
   = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration
   = note: `#[warn(unexpected_cfgs)]` on by default
   = note: this warning originates in the derive macro `EventContent` (in Nightly builds, run with -Z macro-backtrace for more info)
```

### Desired output

```Shell
Not sure whether it's feasible to expect "ruma-macros" instead of "ruma_macros", or it's converted internally and the original name isn't available for the help system.
```

### Rationale and extra context

$ cargo update -p ruma_macros
error: package ID specification `ruma_macros` did not match any packages

help: a package with a similar name exists: `ruma-macros`

That's rather a cosmetic problem, as cargo suggests the proper name later. You may want to fix it or close this issue.

### Other cases

```Rust

```

### Rust Version

```Shell
rustc 1.86.0 (05f9846f8 2025-03-31)
binary: rustc
commit-hash: 05f9846f893b09a1be1fc8560e33fc3c815cfecb
commit-date: 2025-03-31
host: x86_64-unknown-linux-gnu
release: 1.86.0
LLVM version: 19.1.7
```

**Comment by lolbinarycat** [comment]
`ruma_macros` is the *crate name*, this is what rustc sees.

`ruma-macros` is the *package name*, this is what cargo sees.

They may differ arbitrarily, not just by `-` vs `_`.

In order to fix this, we would need either:
* cargo provides rustc with a map of crate names to package names
* cargo intercepts the error/hint and rewrites the crate name to a package name.
